design/one-pager-managed-resource-api-design.md

```diff
@@ -0,0 +1,402 @@
+# Managed Resources API Patterns
+* Owner: Muvaffak Onus (@muvaf)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Terminology
+
+* _External resource_. An actual resource that exists outside Kubernetes, typically in the cloud. AWS RDS or GCP Cloud
+  Memorystore instances are external resources.
+* _Provider_. The entity that owns the physical layer of the external resource. AWS, GCP and Azure are examples of this.
+* _Managed resource_. The Crossplane representation of an external resource. The `RDSInstance` and `CloudMemorystoreInstance`
+  kinds are managed resources.
+* _Claim_. The Crossplane representation of a request for the allocation of a managed resource. Resource claims
+  typically represent the need for a managed resource that implements a particular protocol. MySQLInstance and
+  RedisCluster are examples of resource claims.
+* _Provider_. Cloud provider such as GCP, AWS, Azure offering IaaS, cloud networking, and managed services.
+* _Spec_. The sub-resource of Kubernetes resources that represents the desired state of the user.
+* _Status_ The sub-resource of Kubernetes resources that represents the most up-to-date information about the resource.
+
+For other terms, see [glossary].
+
+## Background
+Crossplane extends Kubernetes heavily and the maintainers always try to adopt the patterns that are already seen in 
+Kubernetes resources, notably declarative style API. However, it's not always intuitive for everyone to see what goes to
+where in a way that is consistent and just as expected by the consumers of those resources.
+
+This document tries to capture how the bridge between Crossplane and provider is shaped. So, the features of managed
+resources that are generic to all of them are not included here, for example reclaim policies. This is more about the
+things that a developer should keep in mind when they try to implement a new managed resource.
+
+A few questions that this document needs to help answering:
+- What goes into `Spec` and what goes into `Status` of a managed resource?
+- Should _all_ information from the provider be represented in the Custom Resource?
+- What parts of the Custom Resource should be allowed to be consumed by the consumers?
+- Do we represent _desired_ state, _observed_ state or both?
+- How should we differentiate the fields that are used by provider from Crossplane's fields?
+
+While this document is heavily opinionated, each provider can wildly vary between each other. So, treat this document as
+a starting point, try to stick to it for consistency but deviate if required and open an issue about why you had to
+deviate so that we can reassess and improve it.
+
+## Guiding Principles
+
+Our users are the stakeholders on how we make decisions. There are two types of users for managed resources:
+* Input Users: Users who configure/create/manage the resource. Crossplane Claim Controller, Cluster Admin are a few
```

Comment:
It could be that "configure" in this case means "write the YAML" (for humans) or "configure the managed resource" (for resource claim based dynamic provisioning). If that were the case, this would already be correctly ordered.